src/wawa-table.ts

```diff
@@ -162,6 +162,33 @@ export class WawaTable extends LitElement {
             <loading-data .loadingTemplate=${this.loadingTemplate}></loading-data>
         </div>`;
     }
+
+    /**
+     * Insert an item into the table at a specific index.
+     * @param item The item to insert into the table
+     * @param index The index to insert the item at
+     */
+    public insertItem(item: any, index: number): void {
+        const newItem: WawaItem = new WawaItem(item, index, this.monitor);
+        this.items.filter(i => i.index >= index).forEach(i => i.index += 1);
+        this.items.splice(index, 0, newItem);
+        this.rows = [];
```

Comment:
It would be better to not clear the rows, but I'll accept it. I'll refactor it when I implement row virtualization.